pkg/controller/controller_instance.go

```diff
@@ -1281,15 +1286,38 @@ func (c *controller) resolveReferences(instance *v1alpha1.ServiceInstance) (*v1a
 		servicePlans, err := c.serviceCatalogClient.ServicePlans().List(listOpts)
 		if err == nil && len(servicePlans.Items) == 1 {
 			sp := &servicePlans.Items[0]
+
+			if sc == nil {
```

Comment:
I think the only way this is nil if it was resolved before and subsequently this became nil? So, this could be for example if the plan gets upgraded, correct?

Reply:
Is there no sane way to utilize the spec.serviceBrokerName in the list function here? I was hoping that we could use that. I guess you'd first look up the SC and then add the broker from there to the selector.

Reply:
Right, it could be nil the field was unset due to the external plan name changing.